src/System.Private.ServiceModel/tests/Common/Infrastructure/xunit/PlatformAttribute.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Xunit;
+using Xunit.Sdk;
+using Xunit.Abstractions;
+
+namespace Infrastructure.Common
+{
+    // The [Platform] attribute can be applied to any test method that
+    // is marked with [WcfFact] or [WcfTheory].  It indicates the which
+    // platforms on which the test should be run.  It also optionally permits
+    // an exclusion list to describe on which platforms it should not be run.
+    // If this attribute is absent, it means the test can run on all platforms.
+    // Examples: 
+    //  [Platform(PlatformID.Windows)]
+    //  [Platform(PlatformID.AnyUnix, Exclude=PlatformID.OSX)]
```

Comment:
How would we use this to specify only exclude OSX? It looks like it would be this:
```csharp
    [Platform(PlatformID.Any, Exclude=PlatformID.OSX)]
```
Can we add a default constructor which sets PlatformID to PlatformID.Any? That way you can specify only the exclude on the attribute which I think is going to be a common use case.